### PR TITLE
Create assignment page (Tutor)

### PR DESCRIPTION
**Pull request by junhonglow**
## Description
Create the frontend page under /assignments/create route that will display a page for tutors to create an assignment. There is currently no limit on the number of questions that can be created for an assignment.

## Related Issue(s)
Partially completes #3, will close after PR #24 is merged as well.

## Screenshots
image

## Additional Notes
1. Added a change to layout, currently if the components exceeded the scroll height of the window, the background would revert back to black.

2. There's a supposed [issue](https://github.com/nextui-org/nextui/pull/2371) that is linked with the input type for date and time for Next UI that has supposedly been fixed in v2.1.17 but not sure why it's still occurring. Either way, I have a workaround but it is not the best looking. 

## TO-DO
- [x] Existing component requires a date time parser for "last modified", if #24 is merged first, I will modify to use util from that PR instead.
- [x] Check if Input for Date and Time is acceptable.

### CODE REVIEW

**Review by WillCWX — CHANGES_REQUESTED** [review]
Good work

There are some comments regarding the handling of inputs.

Do you have a way to handle the submission? It seems that the inputs are not controlled so you can't submit the values directly. There is no overall `<form>` also so I'm not sure how submission will work.

**Comment by junhonglow** [comment]
Issues have been resolved.

File: frontend/src/app/layout.tsx [hunk 1]
```diff
@@ -20,9 +20,9 @@ export default function RootLayout({
     <html lang="en" className="bg-white">
       <body className={inter.className + " min-h-screen"}>
         <Providers>
-          <div className="flex flex-row justify-start">
+          <div className="h-screen flex flex-row justify-start">
```
Comment (WillCWX) [inline frontend/src/app/layout.tsx:23]: i think you have some conflicts with master that vivien just merged in

File: frontend/src/components/forms/AssignmentCreator.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                <Input
+                    className="w-1/3"
+                    label="Deadline"
+                    name="deadline"
+                    onChange={handleInputChange}
+                    type="datetime-local"
+                />
+                <Button className="w-1/3" color="primary" onClick={handleSubmit}>Create Assignment</Button>
+            </div>
+            {renderQuestionUploads()}
```
Comment (WillCWX) [inline frontend/src/components/forms/AssignmentCreator.tsx:139]: This one is ok as it is a render (no react hooks)

File: frontend/src/components/FileUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    Invalid file type.<br />
+                    Please upload a file with one of the following types: {expectedFileTypes.join(",")}
+                </div>
+            );
+        }
+    };
+
+    return (
+        <div>
+            {renderPreview()}
```
Comment (WillCWX) [inline frontend/src/components/FileUpload.tsx:79]: This one is ok as it is a render (no react hooks)

**Review by WillCWX — APPROVED** [review]
Looks good to me!

Just change `{AssignmentCreator()}` to `<AssignmentCreator />` and fix the merge conflicts.

We may need to breakdown the pages into smaller components soon but this should be fine for now.

**Comment by tryyang2001** [comment]
@junhonglow can you fix the build error in frontend? You may see the errors by running `yarn build` locally, and fix those "error" (I think warning is fine for now)

**Pull request merged**